frontend/src/components/MapView/MapTooltip/PopupCharts.tsx

```diff
@@ -0,0 +1,246 @@
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'start',
+    },
+    selectLevelButton: {
+      textTransform: 'none',
+    },
+    selectLevelButtonValue: {
+      display: 'flex',
+      gap: '8px',
+      alignItems: 'center',
+      justifyContent: 'start',
+    },
+    selectLevelButtonText: {
+      overflow: 'hidden',
+      whiteSpace: 'nowrap',
+      textOverflow: 'ellipsis',
+      maxWidth: '280px',
+    },
+    closeButton: {
+      color: 'white',
+      position: 'absolute',
+      right: 0,
+      top: 0,
+    },
+    chartsContainer: {
+      position: 'relative',
+      display: 'flex',
+      flexDirection: 'column',
+      gap: '8px',
+    },
+    charts: {
+      display: 'flex',
+      flexDirection: 'column',
+      gap: '8px',
+    },
+    chartContainer: {
+      display: 'flex',
+      flexDirection: 'column',
+      gap: '8px',
+    },
+    chartSection: {
+      height: '240px',
+      width: '400px',
+      flexGrow: 1,
+    },
+  });
+
+interface PopupChartProps extends WithStyles<typeof styles> {
+  popup: MapTooltipState;
+  setPopupTitle: React.Dispatch<React.SetStateAction<string>>;
+}
+type AdminLevel = 0 | 1 | 2;
+
+const PopupChart = ({ popup, setPopupTitle, classes }: PopupChartProps) => {
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+  const { data } = boundaryLayerData || {};
+  const mapState = useSelector(layersSelector);
+  const dataForCsv = useRef<{ [key: string]: any[] }>({});
+  const [adminLevel, setAdminLevel] = useState<AdminLevel | undefined>(
+    undefined,
+  );
+
+  const adminLevelsNames = useCallback(() => {
+    const locationName = isEnglishLanguageSelected(i18n)
+      ? popup.locationName
+      : popup.locationLocalName;
+    const splitNames = locationName.split(', ');
+
+    const adminLevelLimit =
+      adminLevel === undefined
+        ? MAX_ADMIN_LEVEL
+        : adminLevel + (multiCountry ? 1 : 0);
+    // If adminLevel is undefined, return the whole array
+    // eslint-disable-next-line fp/no-mutating-methods
+    return splitNames.splice(0, adminLevelLimit);
+  }, [adminLevel, popup.locationLocalName, popup.locationName]);
+
+  const startDate1 = new Date().getTime() - oneYearInMs;
+  const endDate1 = new Date().getTime();
+
+  const mapStateIds = mapState.map(item => item.id);
+  const filteredChartLayers = chartLayers.filter(item =>
+    mapStateIds.includes(item.id),
+  );
+
+  const levelsConfiguration = filteredChartLayers.map(item => ({
+    name: item.id,
+    levels: item.chartData?.levels,
+  }));
+
+  useEffect(() => {
+    if (adminLevel !== undefined) {
+      setPopupTitle(adminLevelsNames().join(', '));
+    } else {
+      setPopupTitle('');
+    }
+  }, [adminLevel, adminLevelsNames, setPopupTitle]);
+
+  if (filteredChartLayers.length === 0) {
+    return null;
+  }
+
+  return (
+    <>
+      {adminLevel === undefined ? (
```

Comment:
Using the same component with a ternary feels a bit odd. This means that when you click on the chart, the other data sets that are loaded are still displayed:

<img width="499" alt="Screenshot 2023-11-16 at 8 26 38 PM" src="https://github.com/WFP-VAM/prism-app/assets/16843267/c898f086-d2a5-4133-b622-6c29e55a2e4f">

<img width="420" alt="Screenshot 2023-11-16 at 8 26 33 PM" src="https://github.com/WFP-VAM/prism-app/assets/16843267/dd783a06-3c18-445d-b090-690219582723">

@wadhwamatic is that an ok behavior or should other data be hidden?